src/AngryBytes/Hash/Hash.php

```diff
@@ -110,129 +107,98 @@ public function setSalt($salt)
+    public function shortHash($data)
     {
-        $fullHash = call_user_func_array(
-            array($this, 'hash'),
-            func_get_args()
-        );
-
-        return substr($fullHash, 0, 7);
+        return substr($this->hash($data), 0, 7);
     }
 
     /**
-     * Does something match a short hash?
-     *
-     * First argument to this function is the short hash as it *should* be, all other arguments will
-     * be passed to shortHash()
+     * Verify if the data matches the shortHash
      *
      * @see Hash::shortHash()
      *
-     * @param  mixed  $what
-     * @param  string $hash
+     * @param mixed $data
+     * @param string $shortHash
      * @return bool
      **/
-    public function matchesShortHash()
+    public function verifyShortHash($data, $shortHash)
     {
-        // Full args to method
-        $args = func_get_args();
-
-        // Remove compareTo from the beginning
-        $compareTo = array_shift($args);
-
-        // Create short hash to compare to
-        $shortHash = call_user_func_array(
-            array($this, 'shortHash'),
-            $args
+        return self::compare(
+            $this->shortHash($data),
+            $shortHash
         );
-
-        // Compare and return
-        return self::compare($compareTo, $shortHash);
     }
 
     /**
      * Compare two hashes
      *
-     * This method is time-safe, i.e. it will take the same amount of time to
-     * execute for all inputs. This is critical to avoid timing attacks.
-     *
-     * NOTE:
-     *
-     * **This method only works on ASCII strings.**
+     * Uses the time-save `hash_equals()` function to compare 2 hashes.
      *
      * @param  string $hashA
      * @param  string $hashB
      * @return bool
      **/
     public static function compare($hashA, $hashB)
     {
-        // Compare length
-        if (strlen($hashA) !== strlen($hashB)) {
-            return false;
-        }
-
-        // bitwise OR total for all characters
-        $result = 0;
-
-        for ($charIndex = 0; $charIndex < strlen($hashA); $charIndex++) {
-            // XOR the value of the ASCII characters at $charIndex
-            // This value is then OR-ed into the total
-            $result |= ord($hashA[$charIndex]) ^ ord($hashB[$charIndex]);
-        }
-
-        // If the result is anything but 0 there was a differing character at
-        // one or more of the positions
-        return $result === 0;
+        return hash_equals($hashA, $hashB);
     }
 
     /**
-     * Get the passed arguments as a string
+     * Get the data as a string
      *
-     * Accepts anything serializable
+     * Will serialize non-scalar values
      *
-     * @param mixed $what1
-     * @param mixed $what2
-     * ...
-     * @param  mixed  $whatN
      * @return string
      **/
-    private function getDataString()
+    private function getDataString($data)
     {
-        return serialize(func_get_args());
+        if (is_scalar($data)) {
```

Comment:
Boolean is ook scalar, weet niet of dat goed gaat aangezien het geen string is? Misschien moet je `return (string) $data;` doen voor de zekerheid. Dan wordt true -> 1 en blijft false ''